File: java/detect-sentiment/BasicAuth.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.Base64;

public class BasicAuth {
    public static void main(String args[]) {
        // basic configuration
        String clientId = "[[client_id]]";
        String clientSecret = "[[client_secret]]";

        // create a base64 encoded api key
        String apiKey = Base64.getEncoder().encodeToString((clientId.concat(":")).concat(clientSecret).getBytes());

        // use the api key as a basic token
        String authorization = "Basic ".concat(apiKey);

        // populate endpoint parameters
        String textBody = "[[text_body]]";
        String language = "[[source_language]]";
        String userAgent = "Intelligent API Sample Java Code";

        String jsonPayload = "{ \"text\": \"" + textBody + "\", \"language\": \"" + language + "\" }";

        // invoke the API endpoint
        try {
            HttpClient client = HttpClient.newHttpClient();
            HttpRequest request = HttpRequest.newBuilder()
                    .uri(URI.create("https://api.intelligent-api.com/v1/text/sentiment"))
                    .header("Authorization",
                            authorization)
                    .header("User-Agent",
                            userAgent)
                    .header("Content-Type", "application/json")
                    .POST(HttpRequest.BodyPublishers.ofString(
                            jsonPayload))
                    .build();

            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());

            System.out.println("Response Code: " + response.statusCode());
            System.out.println("Response Body: " + response.body());
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: python/detect-sentiment/basic-auth.py
import base64
import http.client
import json

# basic configuration
client_id = "[[client_id]]"
client_secret = "[[client_secret]]"

# create a base64 encoded api key
api_key_bytes = base64.b64encode((client_id + ":" + client_secret).encode("utf-8"))
api_key = api_key_bytes.decode("utf-8")

# use the api key as a basic token
authorization = "Basic " + api_key

# populate endpoint parameters
text = "[[text_body]]"
language = "[[source_language]]"
user_agent = "Intelligent API Sample Python Code"

data = {
    "text": text,
    "language": language
}

json_data = json.dumps(data)

# invoke the API endpoint
host = "api.intelligent-api.com"
path = "/v1/text/sentiment"

connection = http.client.HTTPSConnection(host)

headers = {
    "Authorization": authorization,
    "Content-Type": "application/json",
    "User-Agent": user_agent
}

connection.request("POST", path, body=json_data, headers=headers)
response = connection.getresponse()
response_data = response.read().decode("utf-8")

print("Status Code:", response.status)
print("Response:", response_data)

connection.close()